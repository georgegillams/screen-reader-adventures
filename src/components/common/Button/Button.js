import React from 'react';
import PropTypes from 'prop-types';
import { Button as GGButton } from 'gg-components/Button';
import nextifyHref from 'client-utils/nextifyHref';
import Link from 'next/link';

const Button = props => {
  const { href, hrefExternal, ...rest } = props;

  const renderNormalButton = !href || hrefExternal;

  const destination = nextifyHref(href);

  if (renderNormalButton) {
    return <GGButton href={href} hrefExternal={hrefExternal} {...rest} />;
  }

  return (
    <Link passHref href={destination.url} as={destination.as} {...destination.options}>
      <GGButton {...rest} />
    </Link>
  );
};

Button.propTypes = {
  href: PropTypes.string,
  hrefExternal: PropTypes.bool,
};

Button.defaultProps = {
  href: null,
  hrefExternal: false,
};

export default Button;
