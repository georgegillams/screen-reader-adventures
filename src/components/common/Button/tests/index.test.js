import React from 'react';

import { render } from '@testing-library/react';

import Button from '..';

describe('<Button />', () => {
  it('Should render correctly', () => {
    const { container } = render(<Button href="/test" />);

    expect(container).toMatchSnapshot();
  });
  it('Should render with nextified href', () => {
    const { container } = render(<Button href="/blog/test" />);

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly with custom className', () => {
    const { container } = render(<Button href="/test" className={'test-custom-classname'} />);

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly hrefExternal', () => {
    const { container } = render(<Button href="https://duckduckgo.com/" hrefExternal />);

    expect(container).toMatchSnapshot();
  });
});
