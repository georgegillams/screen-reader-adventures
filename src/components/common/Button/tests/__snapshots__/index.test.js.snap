// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render correctly 1`] = `
<div>
  <a
    class="button__a button__outer button__outer--regular button__outer--no-bouncy button__outer--link"
    href="/test"
  >
    <span
      class="button__child"
    />
  </a>
</div>
`;

exports[`<Button /> Should render correctly hrefExternal 1`] = `
<div>
  <a
    class="button__a button__outer button__outer--regular button__outer--no-bouncy button__outer--link"
    href="https://duckduckgo.com/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      class="button__child"
    />
  </a>
</div>
`;

exports[`<Button /> Should render correctly with custom className 1`] = `
<div>
  <a
    class="button__a button__outer button__outer--regular button__outer--no-bouncy button__outer--link test-custom-classname"
    href="/test"
  >
    <span
      class="button__child"
    />
  </a>
</div>
`;

exports[`<Button /> Should render with nextified href 1`] = `
<div>
  <a
    class="button__a button__outer button__outer--regular button__outer--no-bouncy button__outer--link"
    href="/blog/test"
  >
    <span
      class="button__child"
    />
  </a>
</div>
`;
